clippy_lints/src/matches/match_like_matches.rs

```diff
@@ -170,3 +180,16 @@ fn find_bool_lit(ex: &ExprKind<'_>) -> Option<bool> {
         _ => None,
     }
 }
+
+fn is_some(path_kind: PatKind<'_>) -> bool {
+    match path_kind {
+        PatKind::TupleStruct(QPath::Resolved(_, path), patterns, _) if is_wild(&patterns[0]) => {
+            let name = path.segments[0].ident;
+            if name.name == rustc_span::sym::Some {
+                return true;
+            }
+            false
```

Comment:
Here you can return the expression result directly, but we should also check for `None`. I believe it would look like this:

```suggestion
            name.name == rustc_span::sym::Some | name.name == rustc_span::sym::None
```